core/src/main/java/aima/core/util/datastructure/Tuple.java

```diff
@@ -0,0 +1,82 @@
+package aima.core.util.datastructure;
+
+/**
+ * Created by @AdrianBZG on 2/04/17.
+ */
+
+import java.util.function.Function;
+
+/**
+ * Util class: a parametric tuple type
+ */
+public class Tuple<A, B> {
+    private A fst;
+    private B snd;
+
+    public Tuple(A fst, B snd) {
+        this.fst = fst;
+        this.snd = snd;
+    }
+
+    public A getWidth() {
+        return fst;
+    }
+
+    public B getHeight() {
```

Comment:
A Tuple does not have a height property. If you need a height, create a `class Dimensions extends Tuple<Width, Height>`

Reply:
or if you want to avoid `Tuple` and use some Java Standard: `class Dimensions extends AbstractMap.SimpleEntry<Width, Height>`